EventAsker.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,28 @@
+using System.Threading.Tasks;
+using EventAsker.API.Dtos;
+using EventAsker.API.Model;
+using EventAsker.API.Repositories;
+using Microsoft.AspNetCore.Mvc;
+
+namespace EventAsker.API.Controllers
+{
+    [Route("api/[controller]")]
+    public class AuthController : Controller
+    {
+        private IAuthRepository _repository;
+        public AuthController(IAuthRepository repository){
+            _repository = repository;
+        }
+        [HttpPost("register")]
+        public async Task<IActionResult> Register([FromBody] AdminForRegistrationDto adminToRegister){
+            //here will be validation if admin with given username already exists, but need to wait for UserExists implementation
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            Admin admin = new Admin(){ Username = adminToRegister.Username };
+            await _repository.Register(admin, adminToRegister.Password);
```

Comment:
It's not a good practice to use repository in controller because repository is a place where you have connection with db - you should consider creating one more layer eg. services. In this case it's also better to pass AdminDto and then map it into Admin.